library/src-3.x/dotty/internal/StringContextMacro.scala

```diff
@@ -1,22 +1,758 @@
+      case tree => QuoteError("Expected statically known argument list", argsExpr)
+    }
+  }
+
+  /** Interpolates the arguments to the formatting String given inside a StringContext
+   *
+   *  @param strCtxExpr the Expr that holds the StringContext which contains all the chunks of the formatting string
+   *  @param args the Expr that holds the sequence of arguments to interpolate to the String in the correct format
+   *  @return the Expr containing the formatted and interpolated String or an error/warning if the parameters are not correct
+   */
+  private def interpolate(strCtxExpr: Expr[StringContext], argsExpr: Expr[Seq[Any]])(implicit reflect: Reflection): Expr[String] = {
+    import reflect._
+    val sourceFile = strCtxExpr.unseal.pos.sourceFile
+
+    val partsExpr = getPartsExprs(strCtxExpr)
+    val args = getArgsExprs(argsExpr)
+
+    val reporter = new Reporter{
+      private[this] var reported = false
+      private[this] var oldReported = false
+      def partError(message : String, index : Int, offset : Int) : Unit = {
+        reported = true
+        val positionStart = partsExpr(index).unseal.pos.start + offset
+        reflect.error(message, sourceFile, positionStart, positionStart)
+      }
+      def partWarning(message : String, index : Int, offset : Int) : Unit = {
+        reported = true
+        val positionStart = partsExpr(index).unseal.pos.start + offset
+        reflect.warning(message, sourceFile, positionStart, positionStart)
+      }
+
+      def argError(message : String, index : Int) : Unit = {
+        reported = true
+        reflect.error(message, args(index).unseal.pos)
+      }
+
+      def strCtxError(message : String) : Unit = {
+        reported = true
+        val positionStart = strCtxExpr.unseal.pos.start
+        reflect.error(message, sourceFile, positionStart, positionStart)
+      }
+      def argsError(message : String) : Unit = {
+        reported = true
+        reflect.error(message, argsExpr.unseal.pos)
+      }
+
+      def hasReported() : Boolean = {
+        reported
+      }
+
+      def resetReported() : Unit = {
+        oldReported = reported
+        reported = false
+      }
+
+      def restoreReported() : Unit = {
+        reported = oldReported
+      }
     }
+
+    interpolate(partsExpr, args, argsExpr, reporter)
   }
 
-}
+  /** Helper function for the interpolate function above
+   *
+   *  @param partsExpr the list of parts enumerated as Expr
+   *  @param args the list of arguments enumerated as Expr
+   *  @param reporter the reporter to return any error/warning when a problem is encountered
+   *  @return the Expr containing the formatted and interpolated String or an error/warning report if the parameters are not correct
+   */
+  def interpolate(partsExpr : List[Expr[String]], args : List[Expr[Any]], argsExpr: Expr[Seq[Any]], reporter : Reporter)(implicit reflect: Reflection) : Expr[String] = {
+    import reflect.{Literal => LiteralTree, _}
```

Comment:
Probably don't need this import

Reply:
This was modified on commit 3e38ef3. I still need to import reflect._ though.

Reply:
Ok